File: backend/src/main/java/com/lifeos/auth/service/AuthService.java
package com.lifeos.auth.service;

import com.lifeos.auth.dto.LoginRequest;
import com.lifeos.auth.dto.RegisterRequest;
import com.lifeos.auth.dto.UserResponse;
import com.lifeos.security.UserPrincipal;
import jakarta.servlet.http.HttpServletResponse;

public interface AuthService {

    UserResponse register(RegisterRequest request, HttpServletResponse response);

    UserResponse login(LoginRequest request, HttpServletResponse response);

    void logout(HttpServletResponse response);

    UserResponse getCurrentUser(UserPrincipal principal);
}

File: frontend/src/services/authService.js
import api from './api';

/**
 * Authentication Service for LifeOS
 * Uses secure HttpOnly cookies (withCredentials: true in api.js).
 * Tokens are never stored in localStorage or sessionStorage.
 */

export const register = async ({ name, email, password }) => {
  const response = await api.post('/auth/register', { name, email, password });
  return response.data;
};

export const login = async ({ email, password }) => {
  const response = await api.post('/auth/login', { email, password });
  return response.data;
};

export const logout = async () => {
  const response = await api.post('/auth/logout');
  return response.data;
};

export const getCurrentUser = async () => {
  const response = await api.get('/auth/me');
  return response.data;
};

export const getCsrfToken = async () => {
  const response = await api.get('/auth/csrf');
  return response.data;
};

export default {
  register,
  login,
  logout,
  getCurrentUser,
  getCsrfToken,
};
